builder/api/src/main/java/io/helidon/builder/api/Prototype.java

```diff
@@ -450,5 +451,39 @@ public interface OptionDecorator<B, T> {
          */
         String[] value();
     }
+
+    /**
+     * Utility class used from generated code for equals and hash code of specific cases of optional.
+     * This exists because of Optional of char arrays, where equals does not work as we need.
+     */
+    public static class OptionalUtil {
+        /**
+         * Equals that uses {@link java.util.Arrays#equals(char[], char[])} in case both optionals have a value.
+         *
+         * @param first  first optional
+         * @param second second optional
+         * @return whether the optionals are equals
+         */
+        public static boolean charArrayEquals(Optional<char[]> first, Optional<char[]> second) {
```

Comment:
Isn't this violating the rule of not using optionals as fields or method parameters ?